src/operator/first.ts

```diff
@@ -5,14 +5,14 @@ import {tryCatch} from '../util/tryCatch';
 import {errorObject} from '../util/errorObject';
 import {EmptyError} from '../util/EmptyError';
 
-export function first<T, R>(predicate?: (value: T, index: number, source: Observable<T>) => boolean,
+export function first<T, R>(predicate?: (value: T, index: number, observable: Observable<T>) => boolean,
                             resultSelector?: (value: T, index: number) => R,
-                            defaultValue?: any): Observable<T> | Observable<R> {
+                            defaultValue?: R): Observable<R> {
```

Comment:
You introduce an additional overload for the method that only includes `T`, but I left those changes out of the first run as it was complicating matters.  Once we get `noImplictAny` and the other type changes in I want to go around and make sure the interfaces get a makeover.